storage/remote/queue_manager.go

```diff
@@ -232,13 +267,14 @@ type QueueManager struct {
 	// https://golang.org/pkg/sync/atomic/#pkg-note-BUG
 	lastSendTimestamp int64
 
-	logger         log.Logger
-	flushDeadline  time.Duration
-	cfg            config.QueueConfig
-	externalLabels labels.Labels
-	relabelConfigs []*relabel.Config
-	client         StorageClient
-	watcher        *wal.Watcher
+	logger          log.Logger
+	flushDeadline   time.Duration
+	cfg             config.QueueConfig
+	externalLabels  labels.Labels
+	relabelConfigs  []*relabel.Config
+	client          StorageClient
+	watcher         *wal.Watcher
```

Comment:
we might want to think about renaming the var here to `walWatcher`, doesn't need to happen now though

Reply:
I'm not keen on that refactor atm, I'll take into account for my next PR.